core/src/main/java/org/apache/brooklyn/util/core/internal/ssh/sshj/SshjTool.java

```diff
@@ -575,9 +585,16 @@ public int execCommands(Map<String,?> props, List<String> commands, Map<String,?
         }
         
         // If async is set, then do it as execScript
-        Boolean execAsync = getOptionalVal(props, PROP_EXEC_ASYNC);
+        Boolean execAsync = getOptionalVal(props, PROP_EXEC_ASYNC, this.execAsync);
         if (Boolean.TRUE.equals(execAsync) && BrooklynFeatureEnablement.isEnabled(BrooklynFeatureEnablement.FEATURE_SSH_ASYNC_EXEC)) {
-            return execScriptAsyncAndPoll(props, commands, env);
+            if (!props.containsKey(PROP_EXEC_ASYNC_POLLING_TIMEOUT.getName())) {
+                return execScriptAsyncAndPoll(
+                        MutableMap.<String, Object>builder().putAll(props).put(PROP_EXEC_ASYNC_POLLING_TIMEOUT.getName(), this.execAsyncPollingTimeout).build(),
+                        commands,
+                        env);
+            } else {
+                return execScriptAsyncAndPoll(props, commands, env);
+            }
```

Comment:
Do the same change for `execScript` as well. Otherwise setting the config on the location will work only for some ssh calls (commands and not scripts).